### ISSUE TITLE
filterByActivity returns NAs for probability

### ISSUE THREAD

**Issue by ajmacdonald3** [MotusDev/Motus-TO-DO#871]
A collaborator working through the R book let me know that the filterByActivity function appears to be returning NAs instead of 1 in the probability column. I tested it out and the R book says the function returns a dataframe though the input is an SQLite connection, but for me it's returning an SQLite connection. When I convert that SQLite connection to a dataframe it does have NAs in the probability column where there should be 1s (the 0s are fine as they're calculated earlier in the function). If I convert the SQLite connection to a dataframe before running the if_else statement, it appears to work fine and the probability column is populated with 1s and 0s. I modified the function as a quick temporary fix, but there may be a more elegant way to code this.

```
# temporary fix to filterByActivity function

filterByActivity_mod <- function (src, return = "good", view = "alltags", 
          minLen = 3, maxLen = 5, maxRuns = 100, ratio = 0.85) 
{
  if (!return %in% c("good", "bad", "all")) {
    stop("'return' must be one of 'good', 'bad', or 'all'", 
         call. = FALSE)
  }
  if (!view %in% c("alltags", "alltagsGPS")) {
    stop("'view' must be one of 'alltags' or 'alltagsGPS'", 
         call. = FALSE)
  }
  if (any(!is.numeric(c(minLen, maxLen, maxRuns, ratio)))) {
    stop("'minLen', 'maxLen', 'maxRuns', and 'ratio' must all be numeric", 
         call. = FALSE)
  }
  if (any(c(minLen, maxLen, maxRuns) < 1)) {
    stop("'minLen', 'maxLen', and 'maxRuns' must all be greater than 0", 
         call. = FALSE)
  }
  if (ratio < 0 | ratio > 1) 
    stop("'ratio' must be a value between 0 and 1", 
         call. = FALSE)
  if (minLen > maxLen) 
    stop("'minLen' must be smaller than or equal to 'maxLen'", 
         call. = FALSE)
  t <- DBI::dbListTables(src$con)
  if (any(!c("runs", "activity", view) %in% t)) {
    stop(paste0("'src' must contain at least tables 'activity', '", 
                view, "', and 'runs'"), call. = FALSE)
  }
  tbl_runs <- dplyr::tbl(src$con, "runs") %>% dplyr::mutate(hourBin = floor(.data$tsBegin/3600))
  tbl_activity <- dplyr::tbl(src$con, "activity")
  if (nrow(DBI::dbGetQuery(src$con, "SELECT * FROM activity LIMIT 1")) < 
      1) {
    stop("'activity' table is empty, cannot filter by activity", 
         call. = FALSE)
  }
  tbl_bad <- dplyr::left_join(tbl_runs, tbl_activity, by = c(batchIDbegin = "batchID", 
                                                             "ant", "hourBin")) %>% dplyr::mutate(run2 = as.numeric(.data$run2), 
                                                                                                  numRuns = as.numeric(.data$numRuns)) %>% dplyr::filter(.data$len < 
                                                                                                                                                           maxLen) %>% dplyr::filter((.data$numRuns >= maxRuns & 
                                                                                                                                                                                        ((.data$run2/.data$numRuns) >= ratio)) | .data$len <= 
                                                                                                                                                                                       minLen) %>% dplyr::select("runID") %>% dplyr::mutate(probability = 0)
  tbl_prob <- dplyr::tbl(src$con, view) %>% dplyr::left_join(tbl_bad, 
                                                             by = "runID") %>%
    collect() %>% as.data.frame() # create dataframe instead of SQLite database connection
  
  tbl_prob <- tbl_prob %>% 
    dplyr::mutate(probability = dplyr::if_else(is.na(.data$probability),
                                               1, .data$probability)) # separate this into two code chunks so probability column is modified after converting to dataframe
  
  if (return == "good") 
    r <- dplyr::filter(tbl_prob, .data$probability > 0)
  if (return == "bad") 
    r <- dplyr::filter(tbl_prob, .data$probability == 0)
  if (return == "all") 
    r <- tbl_prob
  r
}
```

**Comment by plorch** [comment]
This quick fix seems not to be working.  I get an empty table using the following code (from Rbook).  I was able to filter using `is.na(probability)` but there seem to be a large number of records that are neither 0 nor NA.  Any idea what is going on?
```
df.alltags.sub <- tbl.alltags %>% 
  filter(probability == 1) %>%
  collect() %>%
  as.data.frame() %>%
  mutate(ts = as_datetime(ts),  # work with dates AFTER transforming to flat file
         tagDeployStart = as_datetime(tagDeployStart),
         tagDeployEnd = as_datetime(tagDeployEnd))
```

**Comment by steffilazerte** [comment]
Ah, the problem is that the `left_join()` doesn't collect the data anymore and the `if_else()` isn't working at all. I have a fix in the works that should fix this bug, and, hopefully speed things up a bit too. Will keep you posted!

**Comment by steffilazerte** [comment]
Okay this should be fixed in beta4. (Also fixed #884)  To test it out:

```
remotes::install_github("MotusWTS/motus", ref = "beta4")
library(motus)
tags <- tagme(...)
...
```

**Comment by plorch** [comment]
Thanks.  This seems to be fixed also.
